### PR TITLE
Feature 39 product transfer

### PR DESCRIPTION
**Pull request by dimitrov93**

### CODE REVIEW

File: src/components/Button/ProductTransferButton.vue [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    v-if="!item.sold && item.contentOf === 'No'"
+    variant="plain"
+    @click="openDialog"
+  >
+    <v-icon color="#607D8B" size="25">mdi-swap-horizontal</v-icon>
+  </v-btn>
+
+  <v-btn v-else variant="plain" disabled>
+    <v-icon color="#607D8B" size="25">mdi-swap-horizontal</v-icon>
+  </v-btn>
```
Comment (VladoKat) [inline src/components/Button/ProductTransferButton.vue:12]: You can get rid of the second button by:
```suggestion
  <v-btn
    :disabled="item.sold || item.contentOf === 'Yes'"
    variant="plain"
    @click="openDialog"
  >
    <v-icon color="#607D8B" size="25">mdi-swap-horizontal</v-icon>
  </v-btn>
```

File: src/components/Button/ProductTransferButton.vue [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    @close-dialog="closeDialog"
+  />
+</template>
+
+<script setup>
+import ProductTransferDialog from "../Dialog/ProductTransferDialog.vue";
+import { ref } from "vue";
+
+const props = defineProps({
+  item: Object,
```
Comment (VladoKat) [inline src/components/Button/ProductTransferButton.vue:28]: Please, rename this to `product`. `item` is too generic

File: src/components/Dialog/ProductTransferDialog.vue [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    </v-form>
+  </v-dialog>
+</template>
+
+<script setup>
+import { validateUser } from "@/utils/validation-rules";
+import { ref, computed, inject } from "vue";
+import { useStore } from "vuex";
+const snackbarProvider = inject("snackbarProvider");
+const props = defineProps(["modelValue", "item", "userId"]);
```
Comment (VladoKat) [inline src/components/Dialog/ProductTransferDialog.vue:27]: Please, rename `item` to `product` in props

File: src/components/Dialog/ProductTransferDialog.vue
Comment (VladoKat) [inline src/components/Dialog/ProductTransferDialog.vue:1]: `userId` can be removed from this dialog. We can do the fetching of the products/productsForUser in the `ProductTransferButton`'s `closeDialog `method ;)

File: src/views/ProductTransfer.vue
Comment (VladoKat) [inline src/views/ProductTransfer.vue:1]: I guess this is not used? Right?

File: src/router/index.js [hunk 4]
```diff
@@ -88,6 +89,12 @@ const routes = [
     component: Products,
     meta: { title: "Products page", requiresAuth: true },
   },
+  {
+    path: "/products/:productId/transfer",
+    name: "Product Transfer",
+    component: ProductTransfer,
+    meta: { title: "Product Transfer", requiresAuth: true },
+  },
```
Comment (VladoKat) [inline src/router/index.js:97]: We don't use a separate view for the transfer, right? If that's the case, we should remove this.

**Review by VladoKat — CHANGES_REQUESTED** [review]
Generally looks good. There are some things that could be done better. I've noted them in the comments. Reach out if you want to discuss

File: src/components/Button/ProductTransferButton.vue (hunk 1 above)
Comment (dimitrov93) [reply src/components/Button/ProductTransferButton.vue:12]: fixed

File: src/components/Dialog/ProductTransferDialog.vue
Comment (dimitrov93) [reply src/components/Dialog/ProductTransferDialog.vue:1]: yea, but that way everything when we are closing the dialog, we will be fetching and we will do BE fetches, maybe not good idea?

File: src/views/ProductTransfer.vue
Comment (dimitrov93) [reply src/views/ProductTransfer.vue:1]: Correct. Removed

File: src/router/index.js (hunk 4 above)
Comment (dimitrov93) [reply src/router/index.js:97]: removed

File: src/components/Dialog/ProductTransferDialog.vue (hunk 3 above)
Comment (dimitrov93) [reply src/components/Dialog/ProductTransferDialog.vue:27]: done

File: src/components/Button/ProductTransferButton.vue (hunk 2 above)
Comment (dimitrov93) [reply src/components/Button/ProductTransferButton.vue:28]: done

File: src/components/Dialog/ProductTransferDialog.vue [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    </v-form>
+  </v-dialog>
+</template>
+
+<script setup>
+import { validateUser } from "@/utils/validation-rules";
+import { ref, computed, inject } from "vue";
+import { useStore } from "vuex";
+const snackbarProvider = inject("snackbarProvider");
+const props = defineProps(["modelValue", "product", "userId"]);
```
Comment (VladoKat) [inline src/components/Dialog/ProductTransferDialog.vue:34]: Let's keep the way we defineProps consistent. We are defining them differently here compared to the other components:
`const props = defineProps(["modelValue", "product", "userId"]);`
vs
```
const props = defineProps({
modelValue: Boolean, 
product: Object,
userId: String
});
```
Is there a reason for this change here?

**Review by VladoKat — CHANGES_REQUESTED** [review]

File: src/components/Button/ProductTransferButton.vue (hunk 2 above)
Comment (dimitrov93) [reply src/components/Button/ProductTransferButton.vue:27]: i dont know, i changed them everything, not sure why it reversed

File: src/components/Dialog/ProductTransferDialog.vue (hunk 5 above)
Comment (dimitrov93) [reply src/components/Dialog/ProductTransferDialog.vue:34]: updated

**Review by VladoKat — APPROVED** [review]

**Pull request merged**